packages/@aws-cdk/cx-api/lib/cxapi.ts

```diff
@@ -26,8 +26,17 @@ export type CXRequest = ListStacksRequest | SynthesizeRequest;
  */
 export interface MissingContext {
   provider: string;
-  scope: string[];
-  args: string[];
+  props: {
+    account: string;
+    region: string;
+    [key: string]: any;
+  };
+}
+
+export interface HostedZoneProviderProps {
```

Comment:
I feel this should be defined next to the `HostedZoneProvider` class. 

I don't feel comfortable using one struct for both the API properties and defining the wire protocol, because they're distinct things. It's probably okay to have an implicit protocol on the wire (or define 2 structs for 2 distinct purposes).